paimon-common/src/main/java/org/apache/paimon/fs/Path.java

```diff
@@ -62,6 +62,8 @@ public class Path implements Comparable<Path>, Serializable {
     /** A hierarchical URI. */
     private URI uri;
 
+    private boolean isExternalPath;
```

Comment:
We should avoid introducing field in Public API `Path`.

Reply:
done